bin/weewx/drivers/ws1.py

```diff
@@ -51,26 +55,45 @@ def logerr(msg):
 
 class WS1Driver(weewx.drivers.AbstractDevice):
     """weewx driver that communicates with an ADS-WS1 station
-    
-    port - serial port
-    [Required. Default is /dev/ttyS0]
+
+    mode - Communication mode - TCP, UDP, or Serial
+    [Required. Default is serial]
+
+    port - serial port or TCP address
+    [Required. Default is /dev/ttyS0 for serial, and 192.168.36.25:3000 for TCP]
 
     max_tries - how often to retry serial communication before giving up
     [Optional. Default is 5]
 
     retry_wait - how long to wait, in seconds, before retrying after a failure
     [Optional. Default is 10]
+
+    timeout - The amount of time, in seconds, before the connection fails if
+    there is no response
+    [Optional. Default is 3]
     """
     def __init__(self, **stn_dict):
-        self.port = stn_dict.get('port', DEFAULT_PORT)
+        con_mode = stn_dict.get('mode', 'serial').lower()
+        if con_mode == 'serial':
+            self.port = stn_dict.get('port', DEFAULT_SER_PORT)
+        elif con_mode == 'tcp' or con_mode == 'udp':
+            self.port = stn_dict.get(
+                'port', '%s:%d' % (DEFAULT_TCP_ADDR, DEFAULT_TCP_PORT))
+        else:
+            # exit(3)
+            pass
         self.max_tries = int(stn_dict.get('max_tries', 5))
         self.retry_wait = int(stn_dict.get('retry_wait', 10))
         self.last_rain = None
+        timeout = int(stn_dict.get('timeout', 3))
         loginf('driver version is %s' % DRIVER_VERSION)
-        loginf('using serial port %s' % self.port)
+        loginf('using %s port %s' % (con_mode, self.port))
         global DEBUG_READ
         DEBUG_READ = int(stn_dict.get('debug_read', DEBUG_READ))
-        self.station = Station(self.port)
+        if con_mode == 'serial':
+            self.station = StationSerial(self.port, timeout=timeout)
+        elif con_mode == 'tcp' or con_mode == 'udp':
+            self.station = StationInet(self.port, con_mode, timeout=timeout)
```

Comment:
for backward compatibility, please consider this instead:

if con_mode == 'tcp' or con_mode == 'udp':
    self.station = StationInet(...)
else:
    self.station = StationSerial(...)